### PR TITLE
Add support for Postgres LIKE operators

### LINKED ISSUE #8893: Add support for Postgres LIKE operators
### Is your feature request related to a problem or challenge?

Add support for the operators `~~`, `~~*`, `!~~`, `!~~*`, which are short-hand for `LIKE`, `ILIKE`, `NOT LIKE` and `NOT ILIKE` respectively, since they are currently missing.

### Describe the solution you'd like

I've opened a sqlparser-rs PR to enable parsing of these tokens into operator enums: https://github.com/sqlparser-rs/sqlparser-rs/pull/1096

Ideally we'd then just extend the `Operator` enum to encompass the new variants, and add evaluation for them like for other binary operators.

### Describe alternatives you've considered

Rewriting Postgres queries/ASTs to map the operators into the verbatim `LIKE` expressions.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by gruuya**
## Which issue does this PR close?

Closes #8893.

## Rationale for this change

Add support for Postgres `~~`, `~~*`, `!~~`, `!~~*` like-pattern operators, in analogy with the regex-matching operators `~`, `~*`, `!~`, `!~*`.

## What changes are included in this PR?

Extend the `Operator` enum with the new variants, and wire-up evaluation for them.

## Are these changes tested?

Yes.

## Are there any user-facing changes?

New operators available to be used in querying.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -70,7 +70,7 @@ parquet = { version = "50.0.0", default-features = false, features = ["arrow", "
 rand = "0.8"
 rstest = "0.18.0"
 serde_json = "1"
-sqlparser = { version = "0.41.0", features = ["visitor"] }
+sqlparser = { git = "https://github.com/sqlparser-rs/sqlparser-rs", rev = "c7d2903c6d1d1a98e6e6203c5cfd54acb9aeca16", features = ["visitor"] }
```
Comment (alamb) [inline Cargo.toml:73]: I think we need to wait for sqlparser to be released -- see https://github.com/sqlparser-rs/sqlparser-rs/issues/1074  prior to merging this. However I plan to work on releasing sqlparser later today so I think we should be good shortly

File: docs/source/user-guide/sql/operators.md [hunk 2]
```diff
@@ -263,6 +263,58 @@ Not Regex Case-Insensitive Match
 +---------------------------------------------------+
 ```
 
+### `~~`
```
Comment (alamb) [inline docs/source/user-guide/sql/operators.md:266]: ❤️

**Review by alamb — APPROVED** [review]
Thanks @gruuya  -- this looks great to me. I don't think we should merge this PR with a hard coded pin to the sqlparser repository, but as I mentioned I should have a real release of sqlparser shortly

**Comment by alamb** [comment]
https://github.com/apache/arrow-datafusion/pull/9000 has the sqlparser upgrade

File: Cargo.toml (hunk 1 above)
Comment (gruuya) [reply Cargo.toml]: Thanks, that was quick!

I've merged the latest main into this branch now, let me know if there's anything else to do.

**Review by alamb — APPROVED** [review]
Thanks again @gruuya

**Pull request merged**